plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -878,8 +884,33 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
 
         commit(appendFiles, session);
         transaction.commitTransaction();
+        // TODO this may not exactly be the snapshot we committed, if there is another writer
+        long newSnapshotId = transaction.table().currentSnapshot().snapshotId();
         transaction = null;
 
+        // TODO (https://github.com/trinodb/trino/issues/15439): it would be good to publish data and stats atomically
+        previousSnapshotId.ifPresent(previous ->
+                verify(previous != newSnapshotId, "Failed to get new snapshot ID "));
+
+        if (!computedStatistics.isEmpty()) {
+            icebergTable = catalog.loadTable(session, table.getName());
+            beginTransaction(icebergTable);
+            icebergTable = transaction.table();
+            CollectedStatistics collectedStatistics = processCollectedTableStatistics(icebergTable, computedStatistics);
```

Comment:
`processCollectedTableStatistics` looks strange here, method claims to process COLLECTED statistics but you pass COMPUTED statistics in and as a result you get COLLECTED statistics. (caps only to emphasize strange naming)

Reply:
renamed to `processComputedTableStatistics`